Description: Enables using VA-API hardware acceleration in Linux. The patch for bpf_gpu_policy_linux.cc initially came from https://codereview.chromium.org/15955009/diff/92001/content/common/sandbox_linux/bpf_gpu_policy_linux.cc. The code in bpf_gpu_policy_linux.cc was then updated by Steven Newbury to support Nvidia drivers.

Index: dev/media/BUILD.gn
===================================================================
--- dev.orig/media/BUILD.gn
+++ dev/media/BUILD.gn
@@ -373,7 +373,7 @@ component("media") {
     allow_circular_includes_from = [ "//media/base/android" ]
   }
 
-  if (current_cpu != "arm" && is_chromeos) {
+  if (current_cpu != "arm" && (is_chromeos || is_desktop_linux)) {
     sources += [
       "filters/h264_bitstream_buffer.cc",
       "filters/h264_bitstream_buffer.h",
@@ -650,7 +650,7 @@ source_set("unit_tests") {
     }
   }
 
-  if (current_cpu != "arm" && is_chromeos) {
+  if (current_cpu != "arm" && (is_chromeos || is_desktop_linux)) {
     sources += [ "filters/h264_bitstream_buffer_unittest.cc" ]
   }
 
Index: dev/media/gpu/BUILD.gn
===================================================================
--- dev.orig/media/gpu/BUILD.gn
+++ dev/media/gpu/BUILD.gn
@@ -12,7 +12,7 @@ if (is_mac) {
   import("//build/config/mac/mac_sdk.gni")
 }
 
-if (is_chromeos && current_cpu != "arm") {
+if ((is_desktop_linux || is_chromeos) && current_cpu != "arm") {
   action("libva_generate_stubs") {
     extra_header = "va_stub_header.fragment"
 
@@ -238,7 +238,7 @@ component("gpu") {
     }
   }
 
-  if (is_chromeos) {
+  if (is_linux) {
     sources += [
       "accelerated_video_decoder.h",
       "h264_decoder.cc",
@@ -281,7 +281,7 @@ component("gpu") {
         "GLESv2",
       ]
     }
-    if (current_cpu == "arm") {
+    if (current_cpu == "arm" && is_chromeos) {
       sources += [
         "tegra_v4l2_device.cc",
         "tegra_v4l2_device.h",

